datalad_debian/tests/test_update_reprepro_repository.py

```diff
@@ -0,0 +1,132 @@
+import gzip
+from pathlib import Path
+
+from datalad.tests.utils_pytest import (
+    assert_in_results,
+    assert_repo_status,
+    with_tempfile,
+)
+
+from datalad.api import (
+    Dataset,
+    deb_add_distribution,
+    deb_new_distribution,
+    deb_new_package,
+    deb_new_reprepro_repository,
+    deb_update_reprepro_repository,
+    run,
+    save,
+    update,
+)
+
+ckwa = dict(
+    result_renderer='disabled',
+)
+
+
+@with_tempfile
+def test_update_reprepro_repo(path=None):
+    path = Path(path)
+    dist_ds_p = path / 'dist'
+    pkg_ds_p = path / 'dist' / 'packages' / 'tqdm'
+    archive_ds_p = path / 'archive'
+    # prepare a distribution, with a package
+    deb_new_distribution(dist_ds_p, **ckwa)
+    deb_new_package(dataset=dist_ds_p, name=pkg_ds_p.name, **ckwa)
+    # download a tiny package with little to no dependencies
```

Comment:
Not sure if I agree with "no dependencies", as it depends on `python3-all` (and others) to build. ;-)

https://salsa.debian.org/python-team/packages/tqdm/-/blob/master/debian/control
```
Build-Depends: debhelper-compat (= 13),
               dh-python,
               python3-all,
               python3-docopt,
               python3-pandas <!nocheck>,
               python3-pytest <!nocheck>,
               python3-pytest-asyncio <!nocheck>,
               python3-pytest-timeout <!nocheck>,
               python3-rich <!nocheck>,
               python3-setuptools,
               python3-tk <!nocheck>,
```

Perhaps: "download a tiny python package with few dependencies"

Reply:
I removed it. The number of dependencies is irrelevant for this implementation, nothing is built.